--- a/configure.ac	2017-07-19 19:01:08.419235666 +0200
+++ b/configure.ac	2017-07-19 19:01:23.459391913 +0200
@@ -1,9 +1,6 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_PREREQ([2.69])
 AC_CONFIG_MACRO_DIR([m4])
-AC_INIT([psmisc],
-	m4_esyscmd([misc/git-version-gen .tarball-version]),
-	,,,[https://www.gitlab.com/psmisc/psmisc])
 AM_INIT_AUTOMAKE([foreign 1.11 dejagnu subdir-objects -Wall -Wno-portability tar-pax no-dist-gzip dist-xz])
 AM_SILENT_RULES([no])
 AC_CONFIG_SRCDIR([src/comm.h])
