--- a/configure.ac.orig	2017-10-01 13:34:01.003293068 +0200
+++ b/configure.ac	2017-10-01 13:36:41.773291265 +0200
@@ -667,32 +667,10 @@
                     [add_deps_error([gio-unix-2.0])])
 fi
 
-AC_MSG_CHECKING([for glib-networking (GIO TLS implementation)])
 gimp_save_CFLAGS="$CFLAGS"
 gimp_save_LIBS="$LIBS"
-CFLAGS="$CFLAGS $GIO_CFLAGS"
-LIBS="$LIBS $GIO_LIBS"
-AC_RUN_IFELSE([AC_LANG_PROGRAM([#include <gio/gio.h>],
-                               [return !g_tls_backend_supports_tls (g_tls_backend_get_default ());])],
-              [have_glib_networking="yes"],
-              [have_glib_networking="no"],
-              [have_glib_networking="unknown (cross-compiling)"])
 CFLAGS="$gimp_save_CFLAGS"
 LIBS="$gimp_save_LIBS"
-AC_MSG_RESULT($have_glib_networking)
-
-if test "x$have_glib_networking" = "xno"; then
-  add_deps_error([glib-networking],
-      [Test for glib-networking failed. This is required.])
-elif test "x$have_glib_networking" != "xyes"; then
-  warning_glib_networking="
-WARNING: Test for glib-networking cannot be performed while cross-compiling.
-         Make sure glib-networking is installed, otherwise GIMP will not be able
-         to display the remote help pages through the help browser, nor will it
-         be able to open remote HTTPS (or other protocol using SSL/TLS) files.
-         HTTPS is becoming the expected standard and should not be considered
-         optional anymore."
-fi
 
 AC_PATH_PROG(GLIB_COMPILE_RESOURCES, glib-compile-resources, no)
 if test "x$GLIB_COMPILE_RESOURCES" = xno; then
