--- a/.cargo/registry/src/github.com-1ecc6299db9ec823/wlc-sys-0.0.8/build.rs	2017-07-06 15:11:02.445157807 +0200
+++ b/.cargo/registry/src/github.com-1ecc6299db9ec823/wlc-sys-0.0.8/build.rs	2017-07-06 15:13:18.586923394 +0200
@@ -45,13 +45,6 @@
         println!("cargo:rustc-link-lib=dylib=GL");
         println!("cargo:rustc-link-lib=dylib=EGL");
         println!("cargo:rustc-link-lib=dylib=gbm");
-        println!("cargo:rustc-link-lib=dylib=X11");
-        println!("cargo:rustc-link-lib=dylib=xcb");
-        println!("cargo:rustc-link-lib=dylib=xcb-composite");
-        println!("cargo:rustc-link-lib=dylib=xcb-xfixes");
-        println!("cargo:rustc-link-lib=dylib=xcb-xkb");
-        println!("cargo:rustc-link-lib=dylib=X11-xcb");
-        println!("cargo:rustc-link-lib=dylib=xcb-image");
         println!("cargo:rustc-link-lib=dylib=drm");
     } else {
         println!("cargo:rustc-link-lib=dylib=wlc");
@@ -76,6 +69,7 @@
                 .define("WLC_BUILD_STATIC", "ON")
                 .define("WLC_BUILD_EXAMPLES", "OFF")
                 .define("WLC_BUILD_TESTS", "OFF")
+                .define("WLC_X11_SUPPORT", "OFF")
                 .build();
 
     println!("cargo:rustc-link-search=native={}/lib", dst.display());
